java/vector/src/main/java/org/apache/arrow/vector/compression/NoCompressionCodec.java

```diff
@@ -61,6 +61,10 @@ public static class Factory implements CompressionCodec.Factory {
 
     @Override
     public CompressionCodec createCodec(CompressionUtil.CodecType codecType) {
+      if (codecType != CompressionUtil.CodecType.NO_COMPRESSION) {
+        throw new IllegalArgumentException(
+            "Unsupported codec type: " + codecType + " for NoCompressionCodec");
```

Comment:
How about recommending add compression module if codec type is ZSTD_FRAME or LZ4_FRAME? If not, an exception will be thrown.

Reply:
Sounds good. I have refined the error message.